plugins/auth-backend/src/lib/oauth/OAuthAdapter.ts

```diff
@@ -169,11 +179,16 @@ export class OAuthAdapter implements AuthProviderRouteHandlers {
 
       const identity = await this.populateIdentity(response.backstageIdentity);
 
-      // post message back to popup if successful
-      return postMessageResponse(res, appOrigin, {
+      const responseObj: WebMessageResponse = {
         type: 'authorization_response',
         response: { ...response, backstageIdentity: identity },
-      });
+      };
+
+      if (!this.options.isPopupAuthenticationRequest) {
```

Comment:
I feel we don't actually need this option we could just use the redirection if there's a `redirectUrl` in the the state instead. Perhaps it's neat to make that slightly more explicit as well, so we communicate the exact auth mode that's used through state.

Interesting to note as well that we can use the same handler endpoint for all through using communication through state, which is safe to assume that all OAuth implementations have afaik. That means the `/handler/frame` was premature and we could actually migrate back to just a `/handler`. Don't do that in this PR though :grin: